### PR TITLE
Feature/tcvp 1215

### PR DESCRIPTION
**Pull request by lfdgov**
# Description

This PR includes the following proposed change(s):

- {List all the changes, if possible add the jira ticket #}

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Refactoring / Documentation
- [ ] Version change

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Does the change impact or break the Docker build?

- [ ] Yes
- [x] No

If Yes: Has Docker been updated accordingly?

- [ ] Yes
- [ ] No

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/backend/TrafficCourts/Staff.Service/Program.cs [hunk 1]
```diff
@@ -99,8 +99,8 @@
 
 app.UseAuthorization();
 
-app.MapControllers()
-    .RequireAuthorization(); // This will set a default policy that says a user has to be authenticated
+app.MapControllers();
+    // .RequireAuthorization(); // This will set a default policy that says a user has to be authenticated
```
Comment (pbolduc) [inline src/backend/TrafficCourts/Staff.Service/Program.cs:103]: assume this is temporary and wasn't a mistake commit

File: src/frontend/staff-portal/src/app/app.module.ts [hunk 2]
```diff
@@ -100,15 +111,35 @@ export function HttpLoaderFactory(http: HttpClient): TranslateHttpLoader {
...
     {
-      provide: APP_INITIALIZER,
-      useFactory: initializeKeycloak,
-      multi: true,
-      deps: [KeycloakService],
+      provide: Configuration,
+      useFactory: (authService: OidcSecurityService) => new Configuration(
+        {
+          basePath: '',//environment.apiUrl,
+          accessToken: authService.getAccessToken.bind(authService),
```
Comment (pbolduc) [inline src/frontend/staff-portal/src/app/app.module.ts:125]: if you look at the source for the 3rd party oidc, I believe the access token is never used.

File: src/frontend/staff-portal/src/app/services/date.service.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+        } else if (day === '2' || day === '22') {
+            suffix = 'nd';
+        } else if (day === '3' || day === '23') {
+           suffix = 'rd';
+        }
+
+        return suffix;
+
+    }
+}
```
Comment (pbolduc) [inline src/frontend/staff-portal/src/app/services/date.service.ts:21]: we always get these errors about no blank line, try to clean that up. I know it is annoying.

File: src/frontend/staff-portal/src/assets/config/oidc.json [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+    "postLoginRoute": "/ticket",
+    "authority": "https://dev.oidc.gov.bc.ca/auth/realms/ezb8kej4",
+    "redirectUrl": "http://localhost:4200",
+    "postLogoutRedirectUri": "http://localhost:4200",
+    "autoUserInfo": false,
+    "clientId": "tco-staff-portal",
+    "scope": "openid profile",
+    "responseType": "code",
+    "prompt": "consent",
+    "kc_idp_hint": "bceid-basic"
```
Comment (pbolduc) [inline src/frontend/staff-portal/src/assets/config/oidc.json:11]: This is wrong. We dont use bceid, we use idir. This should be `"kc_idp_hint": "idir"`

**Review by pbolduc — CHANGES_REQUESTED** [review]
Need to fix the kc hint to be idir instead of bceid.

File: src/backend/TrafficCourts/Staff.Service/Program.cs (hunk 1 above)
Comment (lfdgov) [reply src/backend/TrafficCourts/Staff.Service/Program.cs:103]: it is temporary till the pull request for 1213

**Comment by KDNeufeld** [comment]
Removal of yarn.lock broke docker-compose up in this PR.
staff-portal no longer builds due to this.

`docker-compose up -d --build staff-portal`
```
Step 7/19 : COPY package.json yarn.lock ./
COPY failed: file not found in build context or excluded by .dockerignore: stat yarn.lock: file does not exist
ERROR: Service 'staff-portal' failed to build : Build failed
```

**Pull request merged**